### PR TITLE
add timestamp for heartbeat

### PR DESCRIPTION
**Pull request by nolouch**
use to report heartbeat interval time.

### CODE REVIEW

File: proto/pdpb.proto [hunk 1]
```diff
@@ -354,6 +354,7 @@ message StoreHeartbeatRequest {
     RequestHeader header = 1;
 
     StoreStats stats = 2;
+    uint64 timestamp = 3;
```
Comment (disksing) [inline proto/pdpb.proto]: `interval` or `duration`? Also please add some comments.

Comment (overvenus) [reply proto/pdpb.proto]: There is a well-known `Timestamp`[1] message.

[1] https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp

**Comment by disksing** [comment]
LGTM.

File: proto/pdpb.proto [hunk 2]
```diff
@@ -242,8 +242,10 @@ message RegionHeartbeatRequest {
     uint64 keys_read = 9;
     // Approximate region size.
     uint64 approximate_size = 10;
-    // Unix timestamp when send the heartbeat.
-    uint64 timestamp = 11;
+    // The unix timestamp in seconds of the start of this period.
+    uint64 start_timestamp = 11;
```
Comment (overvenus) [inline proto/pdpb.proto]: How about moving the `start_timestamp` and the `end_timestamp` into a message?

**Comment by overvenus** [comment]
Cc https://github.com/pingcap/tikv/pull/2965

**Pull request merged**